system_tests/pubsub.py

```diff
@@ -251,13 +262,15 @@ def test_fetch_delete_subscription_w_deleted_topic(self):
         subscription.create()
         topic.delete()
 
-        all_subs = []
-        token = None
-        while True:
-            subs, token = Config.CLIENT.list_subscriptions(page_token=token)
-            all_subs.extend(subs)
-            if token is None:
-                break
+        def _fetch():
+            return list(MethodIterator(Config.CLIENT.list_subscriptions))
+
+        def _found_orphan(result):
+            return any(subscription for subscription in result
+                       if subscription.name == ORPHANED)
```

Comment:
But by using `any`, we could stop retrying before we've actually found **our** orphaned subscription, so long as we've found **an** orphaned subscription